lib/faraday/error.rb

```diff
@@ -74,35 +98,35 @@ class ServerError < Error
   end
 
   # A unified client error for timeouts.
-  class TimeoutError < ServerError
+  class TimeoutError < ClientError
     def initialize(exc = 'timeout', response = nil)
       super(exc, response)
     end
   end
 
   # Raised by Faraday::Response::RaiseError in case of a nil status in response.
   class NilStatusError < ServerError
-    def initialize(_exc, response: nil)
-      message = 'http status could not be derived from the server response'
-      super(message, response)
+    def initialize(exc, response = nil)
```

Comment:
Since 1.0 is not supposed to be compatible with 1.9, we should be free to leave the new syntax in place

Reply:
Normally I'd agree, but I think I made a mistake not insisting on consistency with the other error classes in the original PR.